File: Solutions/[08]_3.cpp
#include <bits/stdc++.h>

using namespace std;

int n;
map<string, int> books;
vector<string> result;

int main() {
	cin >> n;
	for (int i = 0; i < n; i++) {
		string book;
		cin >> book;
		books[book]++;
	}

    int max_value = 0;
	for(auto it = books.begin(); it != books.end(); it++){
		max_value = max(max_value, it->second);
	}
	
	for(auto it = books.begin(); it != books.end(); it++){
		if (it->second == max_value) {
			result.push_back(it->first);
		}
	}
	
	cout << result[0];
}

File: Solutions/[08]_3.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.HashMap;
import java.util.Scanner;

public class Main {
	
	public static void main(String[] args) throws Exception {
		Scanner sc = new Scanner(System.in);
		
		// 정수 이후에 문자열을 받는 경우, 다음과 같이 먼저 문자열로 입력 받아야 합니다.
		int n = Integer.parseInt(sc.nextLine());
		HashMap<String, Integer> books = new HashMap<String, Integer>();
		ArrayList<String> result = new ArrayList<String>();
		int max_value = 0;
		
		for (int i = 0; i < n; i++) {
			String book = sc.nextLine();
			// 해당 책이 출현한 빈도를 계산
			if (books.containsKey(book)) {
				books.put(book, books.get(book) + 1);
			}
			else {
				books.put(book, 1);
			}
			max_value = Math.max(max_value, books.get(book));
		}
		
		// 빈도수가 가장 높은 책의 이름들을 저장
		for (String book: books.keySet()) {
			if (books.get(book) == max_value) result.add(book);
		}
		Collections.sort(result);
		System.out.println(result.get(0));
		
	}

}
